compiler/ras/ILValidationRules.cpp

```diff
@@ -446,6 +446,20 @@ void TR::ValidateChildTypes::validate(TR::Node *node)
          auto childOpcode = node->getChild(i)->getOpCode();
          if (childOpcode.getOpCodeValue() != TR::GlRegDeps)
             {
+				/**
```

Comment:
@Leonardo2718  Can I get review on this change? I just saw the code you recommended, but before that did try following changes and test is passing with it.

Reply:
Isn't this effectively allowing a PassThrough to appear anywhere, as long as the final child is the correct type? My understanding was that it should only be allowed in this one specific case (i.e. under a regStore).

The structure of the code itself looks fine, although I would suggest moving the code bellow the const variable definitions so you can just use those instead of having to call `childOpcode.getDataType().getDataType()`, etc.

Reply:
My understanding for the PassThrough is as it's name suggests, it is just a pass through node and it is OK to have passthrough child as long as real child under passthrough matches the expected child type. Although I can not find anywhere in OMR and OpenJ9 where it uses it like this.

 In https://github.com/eclipse/omr/compare/c20ca06dcf272a1aa1fae1082612c7f5bb8b28c4..e2bb845af7cf4df7240431d864af758e3afee04c Made the changes to keep it limited to PassThrough under regStore only,.

Reply:
In general the compiler will minimize PassThroughs - they are more a transient effect of transformation - if they are allowed to proliferate then we can have a problem because the pattern matching in the compiler will begin to fail. Allowing this only under a regStore makes sense.